build files: add dependency on libgrpc for projects linking libgrpc++

emmenlau opened the pull request:
On MINGW its required that when linking to `libgrpc++`, to also link `libgrpc`. I would assume that this is required for most platforms, but maybe this takes place implicitly? In any case, I have modified build.yaml so that wherever `grpc++` is listed in `dep:`, also `grpc` is listed in `dep:`.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

jtattermusch commented:
please resolve conflicts

jtattermusch commented on build.yaml:
```diff
@@ -3187,8 +3194,8 @@ targets:
   deps:
```
Btw, the order of dependencies in the original code 
https://github.com/grpc/grpc/blob/6b6954050cf0379dbbee90dd18313a3779e0dd52/build.yaml#L3204 (it should be mostly the reverse, as for all the other targets).

emmenlau commented:
I have fixed the conflicts, and changed to order of dependencies. To fix the conflicts, I hopefully
did it in a clean way: while merging I regenerated the auto-generated build files and overwrote
the conflicting files, then commited the updated merge. Is that ok/good practice?

jtattermusch commented:
@emmenlau  yes, rewriting the generated files is a good practice.

jtattermusch commented on build.yaml:
```diff
@@ -1230,8 +1235,8 @@ libs:
   - grpc_test_util
-  - grpc++
   - grpc
+  - grpc++
```
The dependencies in build.yaml need to come in reverse order. (grpc++  comes before grpc)
(many occurences).

jtattermusch commented:
Is this still needed?  (also, there are merge conflicts that would need to be resolved).
Closing for now, feel free to reopen.

emmenlau commented:
Closed in favor of new PR https://github.com/grpc/grpc/pull/10075
